{"ast":null,"code":"var _jsxFileName = \"C:\\\\Users\\\\Edi\\\\Desktop\\\\MyWebsite\\\\src\\\\ErrorBoundary.js\";\n// ErrorBoundary.js\nimport React from 'react';\nimport { jsxDEV as _jsxDEV } from \"react/jsx-dev-runtime\";\nclass ErrorBoundary extends React.Component {\n  constructor(props) {\n    super(props);\n    this.state = {\n      hasError: false\n    };\n  }\n  static getDerivedStateFromError() {\n    // Update state so the next render shows the fallback UI.\n    return {\n      hasError: true\n    };\n  }\n  componentDidCatch(error, info) {\n    // Log the error to an error reporting service or console.\n    console.error('Error caught by boundary:', error, info);\n  }\n  render() {\n    if (this.state.hasError) {\n      // You can customize the fallback UI here.\n      return /*#__PURE__*/_jsxDEV(\"div\", {\n        className: \"text-center mt-5\",\n        children: [/*#__PURE__*/_jsxDEV(\"h2\", {\n          children: \"Something went wrong.\"\n        }, void 0, false, {\n          fileName: _jsxFileName,\n          lineNumber: 25,\n          columnNumber: 11\n        }, this), /*#__PURE__*/_jsxDEV(\"p\", {\n          children: \"We're sorry for the inconvenience. Please try again later.\"\n        }, void 0, false, {\n          fileName: _jsxFileName,\n          lineNumber: 26,\n          columnNumber: 11\n        }, this)]\n      }, void 0, true, {\n        fileName: _jsxFileName,\n        lineNumber: 24,\n        columnNumber: 9\n      }, this);\n    }\n    return this.props.children;\n  }\n}\nexport default ErrorBoundary;","map":{"version":3,"names":["React","jsxDEV","_jsxDEV","ErrorBoundary","Component","constructor","props","state","hasError","getDerivedStateFromError","componentDidCatch","error","info","console","render","className","children","fileName","_jsxFileName","lineNumber","columnNumber"],"sources":["C:/Users/Edi/Desktop/MyWebsite/src/ErrorBoundary.js"],"sourcesContent":["// ErrorBoundary.js\r\nimport React from 'react';\r\n\r\nclass ErrorBoundary extends React.Component {\r\n  constructor(props) {\r\n    super(props);\r\n    this.state = { hasError: false };\r\n  }\r\n\r\n  static getDerivedStateFromError() {\r\n    // Update state so the next render shows the fallback UI.\r\n    return { hasError: true };\r\n  }\r\n\r\n  componentDidCatch(error, info) {\r\n    // Log the error to an error reporting service or console.\r\n    console.error('Error caught by boundary:', error, info);\r\n  }\r\n\r\n  render() {\r\n    if (this.state.hasError) {\r\n      // You can customize the fallback UI here.\r\n      return (\r\n        <div className=\"text-center mt-5\">\r\n          <h2>Something went wrong.</h2>\r\n          <p>We're sorry for the inconvenience. Please try again later.</p>\r\n        </div>\r\n      );\r\n    }\r\n\r\n    return this.props.children;\r\n  }\r\n}\r\n\r\nexport default ErrorBoundary;\r\n"],"mappings":";AAAA;AACA,OAAOA,KAAK,MAAM,OAAO;AAAC,SAAAC,MAAA,IAAAC,OAAA;AAE1B,MAAMC,aAAa,SAASH,KAAK,CAACI,SAAS,CAAC;EAC1CC,WAAWA,CAACC,KAAK,EAAE;IACjB,KAAK,CAACA,KAAK,CAAC;IACZ,IAAI,CAACC,KAAK,GAAG;MAAEC,QAAQ,EAAE;IAAM,CAAC;EAClC;EAEA,OAAOC,wBAAwBA,CAAA,EAAG;IAChC;IACA,OAAO;MAAED,QAAQ,EAAE;IAAK,CAAC;EAC3B;EAEAE,iBAAiBA,CAACC,KAAK,EAAEC,IAAI,EAAE;IAC7B;IACAC,OAAO,CAACF,KAAK,CAAC,2BAA2B,EAAEA,KAAK,EAAEC,IAAI,CAAC;EACzD;EAEAE,MAAMA,CAAA,EAAG;IACP,IAAI,IAAI,CAACP,KAAK,CAACC,QAAQ,EAAE;MACvB;MACA,oBACEN,OAAA;QAAKa,SAAS,EAAC,kBAAkB;QAAAC,QAAA,gBAC/Bd,OAAA;UAAAc,QAAA,EAAI;QAAqB;UAAAC,QAAA,EAAAC,YAAA;UAAAC,UAAA;UAAAC,YAAA;QAAA,OAAI,CAAC,eAC9BlB,OAAA;UAAAc,QAAA,EAAG;QAA0D;UAAAC,QAAA,EAAAC,YAAA;UAAAC,UAAA;UAAAC,YAAA;QAAA,OAAG,CAAC;MAAA;QAAAH,QAAA,EAAAC,YAAA;QAAAC,UAAA;QAAAC,YAAA;MAAA,OAC9D,CAAC;IAEV;IAEA,OAAO,IAAI,CAACd,KAAK,CAACU,QAAQ;EAC5B;AACF;AAEA,eAAeb,aAAa","ignoreList":[]},"metadata":{},"sourceType":"module","externalDependencies":[]}